collections/collections.go

```diff
@@ -0,0 +1,94 @@
+package collections
+
+import (
+	"errors"
+	"math"
+
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+)
+
+var (
+	// ErrNotFound is returned when the provided key is not present in the StorageProvider.
+	ErrNotFound = errors.New("collections: not found")
+	// ErrEncoding is returned when something fails during key or value encoding/decoding.
+	ErrEncoding = errors.New("collections: encoding error")
+)
+
+// StorageProvider is implemented by types
+// which provide a KVStore given a StoreKey.
+// It represents sdk.Context, it exists to
+// reduce dependencies.
+type StorageProvider interface {
```

Comment:
We should actually just use `context.Context` in `Insert`, `Update`, etc. That is most consistent with the direction things are going. Users shouldn't need to use `sdk.UnwrapSDKContext` at all. The collection types should do that for them

Reply:
addressed as discussed